### PR TITLE
Bolt v2: Refactor and Consolidate Doc Site

### PR DESCRIPTION
**Pull request by sghoweri**
## Summary
In a nutshell, this PR is part 2 of the docs site refactor work @mikemai2awesome started got the ball rolling on, cleaning up the site and reorganizing some things to help us better maintain the Docs site-specific codebase.

### CODE REVIEW

File: apps/bolt-site/components/layout/layout.scss [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      grid-column-end: 4;
+    }
+  }
+}
+
+.c-bds-layout__header__nav {
+  padding: 15px 20px;
+  text-align: right;
+
+  @media (max-width: 599px) {
```
Comment (mikemai2awesome) [inline apps/bolt-site/components/layout/layout.scss]: `@include bolt-mq($until: small)` should be used here.

**Comment by mikemai2awesome** [comment]
@sghoweri 

White spaces are removed in unexpected places:

screen shot 2018-09-17 at 3 38 33 pm
screen shot 2018-09-17 at 3 39 11 pm

Open menu text is appearing in the mobile hamburger menu.

screen shot 2018-09-17 at 3 38 50 pm

**Comment by sghoweri** [comment]
> Open menu text is appearing in the mobile hamburger menu.
> 
> screen shot 2018-09-17 at 3 38 50 pm

@mikemai2awesome @remydenton this was actually how the separate 
https://github.com/bolt-design-system/bolt/pull/869 PR popped up — I tried to use the Button component to build the new Floating Action Button and saw there was an unrelated issue to the Button component’s Icon-only / Rounded options!

**Comment by mikemai2awesome** [comment]
> @mikemai2awesome I’m a bit stumped on the weird whitespace / text behavior... have you seen this sort of thing before?

Do we have spaceless somewhere? @sghoweri

**Comment by sghoweri** [comment]
Thanks @mikemai2awesome!

**Pull request merged**